5.6/docker-entrypoint.sh

```diff
@@ -48,14 +48,17 @@ if [ "$1" = 'mysqld' ]; then
 			-- What's done in this file shouldn't be replicated
 			--  or products like mysql-fabric won't work
 			SET @@SESSION.SQL_LOG_BIN=0;
-			
+
 			DELETE FROM mysql.user ;
 			CREATE USER 'root'@'%' IDENTIFIED BY '${MYSQL_ROOT_PASSWORD}' ;
 			GRANT ALL ON *.* TO 'root'@'%' WITH GRANT OPTION ;
 			DROP DATABASE IF EXISTS test ;
 			FLUSH PRIVILEGES ;
 		EOSQL
-		mysql+=( -p"${MYSQL_ROOT_PASSWORD}" )
+
+		if [ -z "$MYSQL_ALLOW_EMPTY_PASSWORD" ]; then
```

Comment:
This needs to just be `-z "$MYSQL_ROOT_PASSWORD"` or else we'll fail on things like `-e MYSQL_ALLOW_EMPTY_PASSWORD=1 -e MYSQL_ROOT_PASSWORD=something` (which currently works).

We already bail above on `MYSQL_ALLOW_EMPTY_PASSWORD`, so at this point if `MYSQL_ROOT_PASSWORD` is empty, we have an empty password.